api/app/helpers/application_helper.rb

```diff
@@ -11,4 +11,74 @@ def json(data, include: nil, except: nil)
         end
         data
     end
+
+    def filter_given_id(table, filter_id, custom_table_data = true)
+        table_data = custom_table_data ? table : table.order(:id)
+        table_data.select do |entry|
+            entry[filter_id] == params[filter_id].to_i
+        end
+    end
+
+    def index_response(table, filter_table, filter_func, custom_table_data = false)
+        filter_id = table_to_id(filter_table)
+        unless params.include?(filter_id)
+            render_success(custom_table_data ? table : table.order(:id))
+            return
+        end
+        return if invalid_id(filter_table, filter_id)
+
+        render_success(filter_func)
+    end
+
+    def table_to_id(table)
+        "#{table.to_s.gsub(/([a-z])([A-Z])/, '\1_\2').downcase}_id".to_sym
```

Comment:
This really scares me...I don't like auto conversion. I would rather have a dictionary lookup.